### PR TITLE
Add compose_object() API

### PR DESCRIPTION
**Pull request by balamurugana**
Signed-off-by: Bala.FA <bala@minio.io>

### CODE REVIEW

File: tests/tests.rs [hunk 1]
```diff
@@ -498,6 +498,92 @@ impl<'a> ClientTest<'_> {
...
+
+        let object_name = rand_object_name();
+
+        self.client
+            .compose_object(
+                &mut ComposeObjectArgs::new(&self.test_bucket, &object_name, &mut sources).unwrap(),
+            )
+            .await
+            .unwrap();
+
```
Comment (harshavardhana) [inline tests/tests.rs:576]: Can you check head object to ensure the right object size is present on the server ?

File: tests/tests.rs [hunk 2]
```diff
@@ -498,6 +498,92 @@ impl<'a> ClientTest<'_> {
...
+                &CopyObjectArgs::new(
+                    &self.test_bucket,
+                    &object_name,
+                    CopySource::new(&self.test_bucket, &src_object_name).unwrap(),
+                )
+                .unwrap(),
+            )
+            .await
+            .unwrap();
+
```
Comment (harshavardhana) [inline tests/tests.rs:532]: Better to verify the size of the object after copy..

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: tests/tests.rs (hunk 2 above)
Comment (balamurugana) [reply tests/tests.rs:532]: Done

File: tests/tests.rs (hunk 1 above)
Comment (balamurugana) [reply tests/tests.rs:583]: Done

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**